internal/pkg/manifest/workload_ecs.go

```diff
@@ -299,19 +299,37 @@ func (lc *Logging) GetEnableMetadata() *string {
 
 // SidecarConfig represents the configurable options for setting up a sidecar container.
 type SidecarConfig struct {
-	Port          *string              `yaml:"port"`
-	Image         *string              `yaml:"image"`
-	Essential     *bool                `yaml:"essential"`
-	CredsParam    *string              `yaml:"credentialsParameter"`
-	Variables     map[string]Variable  `yaml:"variables"`
-	Secrets       map[string]Secret    `yaml:"secrets"`
-	MountPoints   []SidecarMountPoint  `yaml:"mount_points"`
-	DockerLabels  map[string]string    `yaml:"labels"`
-	DependsOn     DependsOn            `yaml:"depends_on"`
-	HealthCheck   ContainerHealthCheck `yaml:"healthcheck"`
+	Port          *string                            `yaml:"port"`
+	Image         Union[*string, SidecarImageConfig] `yaml:"image"`
+	Essential     *bool                              `yaml:"essential"`
+	CredsParam    *string                            `yaml:"credentialsParameter"`
+	Variables     map[string]Variable                `yaml:"variables"`
+	Secrets       map[string]Secret                  `yaml:"secrets"`
+	MountPoints   []SidecarMountPoint                `yaml:"mount_points"`
+	DockerLabels  map[string]string                  `yaml:"labels"`
+	DependsOn     DependsOn                          `yaml:"depends_on"`
+	HealthCheck   ContainerHealthCheck               `yaml:"healthcheck"`
 	ImageOverride `yaml:",inline"`
 }
 
+// SidecarImageConfig represents the docker build arguments and location of the existing image.
+type SidecarImageConfig struct {
```

Comment:
sorry I am late to the party 😂 

This is not a request to change the implementation but just an idea. What do you think of doing this:
```go
type ImageLocationOrBuild struct {
	Build    Union[*string, DockerBuildArgs] `yaml:"build"`
	Location *string                                         `yaml:"location"`
}
type Image struct {
      ImageLocationOrBuild `yaml:",inline"`
}

type SidecarConfig struct {
     Image         Union[*string, ImageLocationOrBuild] `yaml:"image"`
}
```

This struct here is named `SidecarImageConfig` but there is actually nothing specific about `Sidecar` in it. It looks like a general purpose struct that can be used for main container as well.